### PR TITLE
feat (3dtiles): use different subdiv. logic for add and sub refinement

### PR DESCRIPTION
**Pull request by peppsac**
This commit changes the way additive refinement is handled for 3dtiles.
Until now we had a single option: when a tile reached its sse threshold,
we would download and display all of its children.

In the additive refinement we can do better and only download the children
that will be visible. This reduces network/mem usage, for the same visual
result.

### CODE REVIEW

File: src/Process/3dTilesProcessing.js [hunk 1]
```diff
@@ -7,41 +9,86 @@ function requestNewTile(view, scheduler, geometryLayer, metadata, parent) {
...
-
         Promise.all(promises).then(() => {
             node.pendingSubdivision = false;
             context.view.notifyChange(true);
         });
     }
 }
 
-export function $3dTilesCulling(node, camera) {
+export function $3dTilesCulling(node, camera, overrideMatrixWorld) {
```
Comment (Jeremy-Gaillard) [inline src/Process/3dTilesProcessing.js]: Not a big fan of this override. I would prefer something like this instead: 
```js
export function $3dTilesCulling(viewerRequestVolume, boundingVolume, matrixWorld, camera)
```

**Review by Jeremy-Gaillard — CHANGES_REQUESTED** [review]
Very quick review, but this seems nice. It also fixes the issue with the viewerRequestVolume not being properly handled.

File: src/Process/3dTilesProcessing.js (hunk 1 above)
Comment (peppsac) [reply src/Process/3dTilesProcessing.js]: I changed the function, and fixed the special case of boundingVolume.region. Let me know what you think.

File: src/Core/Scheduler/Providers/3dTiles_Provider.js [hunk 2]
```diff
@@ -170,21 +191,18 @@ function configureTile(tile, layer, metadata, parent) {
...
         tile.additiveRefinement = parent ? (parent.additiveRefinement) : false;
     }
-    tile.parentFromLocalTransform = tile.transform;
-    tile.worldFromLocalTransform = new THREE.Matrix4().multiplyMatrices(parent ? parent.worldFromLocalTransform : new THREE.Matrix4(), tile.parentFromLocalTransform);
-    const m = new THREE.Matrix4();
-    m.getInverse(tile.worldFromLocalTransform);
-    tile.viewerRequestVolume = metadata.viewerRequestVolume ? getBox(metadata.viewerRequestVolume, m) : undefined;
-    tile.boundingVolume = getBox(metadata.boundingVolume, m);
+    tile.viewerRequestVolume = metadata.viewerRequestVolume;
+    tile.boundingVolume = metadata.boundingVolume;
```
Comment (Jeremy-Gaillard) [inline src/Core/Scheduler/Providers/3dTiles_Provider.js:214]: Can you remove these? It is confusing to have the same information stored at two different places. I'd suggest only keeping a link to the metadata.

**Review by Jeremy-Gaillard — CHANGES_REQUESTED** [review]
You forgot to change the call to recurse in the extendTileset function. It should be something like this:
```js
recurse(tileset.root, baseURL, this.index[nodeId]);
```

File: src/Core/Scheduler/Providers/3dTiles_Provider.js (hunk 2 above)
Comment (peppsac) [reply src/Core/Scheduler/Providers/3dTiles_Provider.js:214]: Both prperties are used often, so removing them *could* have some perf impact.
I'd prefer to not remove them for now.

**Comment by peppsac** [comment]
Fixed the `recurse`call.

File: src/Core/Scheduler/Providers/3dTiles_Provider.js (hunk 2 above)
Comment (Jeremy-Gaillard) [reply src/Core/Scheduler/Providers/3dTiles_Provider.js:214]: That not exactly what I meant. My point was there should be a clearer separation between the metadata part of the object (bounding volume, etc.) and Object 3D part (transformation). Presently, both are mixed and sometimes we call culling on the metadata part, sometimes on the Object 3D part.

**Review by Jeremy-Gaillard — APPROVED** [review]

**Pull request merged**